src/service/history-impl.js

```diff
@@ -109,6 +112,22 @@ export class History {
   }
 
   /**
+   * Helper method to handle navigation to a local target, e.g. When a user clicks an
+   * anchor link to a local hash - <a href="#section1">Go to section 1</a>.
+   *
+   * @param {string} target
+   * @return {!Promise}
+   */
+  replaceStateForTarget(target) {
+    const previousHash = this.ampdoc_.win.location.hash;
+    return this.push(() => {
+      this.ampdoc_.win.location.replace(previousHash || '#');
+    }).then(() => {
```

Comment:
Not sure where is the 2nd block, but if you mean inside the push that's the wrong place to have it, the push argument is a callback to be called when popping that state.